contracts/OffsetHelper.sol

```diff
@@ -296,102 +344,90 @@ contract OffsetHelper is OffsetHelperStorage {
     }
 
     /**
-     * @notice Checks whether an address can be used by the contract.
-     * @param _erc20Address address of the ERC20 token to be checked
-     * @return True if the address can be used by the contract
+     * @notice Redeems the specified amount of NCT / BCT for TCO2.
+     * @dev Needs to be approved on the client side
+     * @param _fromToken Could be the address of NCT
+     * @param _amount Amount to redeem
+     * @return tco2s An array of the TCO2 addresses that were redeemed
+     * @return amounts An array of the amounts of each TCO2 that were redeemed
      */
-    function isEligible(address _erc20Address) private view returns (bool) {
-        bool isToucanContract = IToucanContractRegistry(contractRegistryAddress)
-            .checkERC20(_erc20Address);
-        if (isToucanContract) return true;
-        if (_erc20Address == eligibleTokenAddresses["BCT"]) return true;
-        if (_erc20Address == eligibleTokenAddresses["NCT"]) return true;
-        if (_erc20Address == eligibleTokenAddresses["USDC"]) return true;
-        if (_erc20Address == eligibleTokenAddresses["WETH"]) return true;
-        if (_erc20Address == eligibleTokenAddresses["WMATIC"]) return true;
-        return false;
-    }
+    function autoRedeem(
+        address _fromToken,
+        uint256 _amount
+    )
+        public
+        onlyRedeemable(_fromToken)
+        returns (address[] memory tco2s, uint256[] memory amounts)
+    {
+        require(
+            balances[msg.sender][_fromToken] >= _amount,
+            "Insufficient NCT/BCT balance"
+        );
 
-    /**
-     * @notice Checks whether an address can be used in a token swap
-     * @param _erc20Address address of token to be checked
-     * @return True if the specified address can be used in a swap
-     */
-    function isSwappable(address _erc20Address) private view returns (bool) {
```

Comment:
The diffs are so confusing here, where are these functions going?

Reply:
not removed. I just re-organized the functions because there was no structure in the file for public, internal and private functions. 


only happened in the second last commit.